go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -643,3 +644,27 @@ func AddRoutesGatewayIP(
 
 	return nil
 }
+
+func shouldSkipAllocateIPsError(err error, networkAllocated bool, ipamClaim *ipamclaimsapi.IPAMClaim) bool {
+	// Only skip if it's an "already allocated" error
+	if !ip.IsErrAllocated(err) {
+		return false
```

Comment:
this is for backwards compatibility? why were we ignoring allocation errors? - before if its ignored then.. what did we do?

Reply:
Before this change, IPAllocator _could_ detect IP duplications, but we _always_ ignored it.
The reasons it was ignored were: 
1. current controller behavior is that a pod created get's reconciled multiple times (once at creation where it gets the IP allocated and get no err, then on every pod update it would get `ipAllocErr` although it's the same pod using the IP) so in order to prevent false-positive IP duplication errors - the err was just ignored.
2. before static-ips for primary UDN feature - IPs were always allocated by OVNK so there was a very low chance of duplication. Now that we introduced static-ips - we need to account for human errs.

Reply:
thank you for outlining the reasons.